boggle_blast.lic

```diff
@@ -0,0 +1,165 @@
+    @danger_items = ['doll'] # todo: add this in to trash_items?
+    @fake_pumpkins = ['wooden pumpkins', 'cambrinth pumpkin']
+    do_boggle_crap
+  end
+
+  def do_boggle_crap
+    loop do
+      heal_up
+      stow_hands
+      ditch_pumpkins
+      empty_hands
+      check_money
+      go_play
+      check_prize
+    end
+  end
+
+  def heal_up
+    check_health
+    if bleeding?
+      walk_to('19104')
+      bput("go path", "Obvious paths: none")
+      pause 5 # assumes quick spider travel option purchased
+      wait_for_script_to_complete('safe-room')
+      pause
+      walk_to(@portal_room)
+      bput("go path", "Obvious paths: none")
+      pause 5 # assumes quick spider travel option purchased
+      walk_to('10584')
+      pause
+    end
+  end
+
+  def check_money
+    get_money if check_wealth('dokoras') < 200 && @withdraw
+  end
+
+  def get_money
+    walk_to('10358')
+    case bput("withdraw #@cash #@denomination #@currency ", 'The clerk counts out', 'The clerk tells you', 'You must be at')
+    when "The clerk counts out"
+      echo "got cash"
+      return
+    else
+      echo "Out of cash!"
+      exit
+    end
+  end
+
+  def go_play
+    walk_to('10584')
+    unless bput('give farmer 200 dokoras', 'The farmer takes your money', 'Sorry, the game rooms are all full right now') == "The farmer takes your money"
+      while line = get
+        if line =~ /^The pumpkin farmer (beams happily|grins) at/
+          response = bput('give farmer 200 dokoras', 'The farmer takes your money', 'Sorry, the game rooms are all full right now')
+          next if response =~ /Sorry, the game rooms are all full right now/
+          break if response =~ /The farmer takes your money/
+        end
+      end
+    end
+    5.times do
+      @true_pumpkin = false
+      get_pumpkin while !@true_pumpkin
+      case bput('load slingshot', 'You carefully load', "You need to have a pumpkin")
+      when "You need to have a pumpkin"
+        stow_hands
+        @true_pumpkin = false
+        get_pumpkin while !@true_pumpkin
+        bput('load slingshot', 'You carefully load')
+      end
+      case bput('fire slingshot', 'You position the slingshot', 'You let loose', 'You carefully aim the', 'You really should be standing to play', 'You fire the slingshot and awkwardly', 'You take careful aim')
+      when 'You really should be standing to play'
+        bput('stand', 'You stand back up')
+        bput('fire slingshot', 'You position the slingshot', 'You let loose', 'You carefully aim the', 'You fire the slingshot and awkwardly', 'You take careful aim')
+      end
+      pause
+    end
+    pause
+    case bput("go door", "Obvious exits", "You must be standing to do that")
+    when "You must be standing to do that"
+      bput('stand', 'You stand back up')
```

Comment:
💡 There's a `stand()` helper that will handle a few edge cases and make this more reliable

Reply:
oh sweet! I'll check it out

Reply:
I just realized the helper is actually named [fix_standing](https://github.com/rpherbig/dr-scripts/blob/master/common.lic#L261) - sorry for any confusion